### ISSUE TITLE
Customization of template invoice2.blade.php

### ISSUE THREAD

**Issue by BIIINO** [crater-invoice/crater#895]
Hello,

I am trying to little bit customize invoice2.blade.php when i wan to just switch between shipping address and billing address.

So i want to have billing address on the right side and shipping on the left - in the middle between company address and customer billing address.  
![image](https://user-images.githubusercontent.com/60783519/163628417-df2ef0a8-71c5-45ac-a4f6-cb336f091a4c.png)

I really spent lot of time how to change it but without success.

Please could you help me how to do this ? I think is not big deal but i cant find the right code in template.

Thanks in advance

**Comment by enoversum** [comment]
Everything is really in the templates, both the CSS as well as the corresponding HTML and blade PHP code. It is even named in a way so you could understand what element does what in there. You can also look into the html file and search for 'billing' in order to find the right place in the file, and exchange the code for billing and invoice addresses.

**Comment by BIIINO** [comment]
@enoversum hi i understand. I allready did lot of customization of others templates but that i cant figure out what is the right segment of css

**Comment by enoversum** [comment]
@BIIINO One thing is the CSS, the other is the actual HTML/PHP that you can change. I used another solution for invoices before and rebuilt my template for Crater, based on the standard template, so it's possible to change almost every aspect of your template; you just need to find the right place in the code and can exchange places for billing and shipping addresses, for example.
One disadvantage of Crater is that you cannot see an HTML version of your invoice, so it's easier to find the code you need to change, but the code itself should help you already, as there are blade functions that wear the name of what is displayed. The CSS is at the top and its classes start with `.billing-`, shipping and billing/invoice details are here, as the naming suggests: 

`@if ($shipping_address !== '</br>')
                <div class="shipping-address-container shipping-address">
                    @if ($shipping_address)
                        <b>@lang('pdf_ship_to')</b> <br>
                        {!! $shipping_address !!}
                    @endif
                </div>
            @endif


            <div class="billing-address-container billing-address" @if ($shipping_address === '</br>') style="float:right; margin-right:30px;" @endif>
                @if ($billing_address)
                    <b>@lang('pdf_bill_to')</b> <br>
                    {!! $billing_address !!}
                @endif
            </div>`

**Comment by BIIINO** [comment]
Yes also i found that, but i was / am not able to switch positions between these two addresses when i change any part of that html or css code.

**Comment by enoversum** [comment]
@BIIINO I'm sorry, that sounds like you need someone who is fluent in CSS and HTML/Blade to help you. I am just another user like you and no official support. I suspect the struggle comes from CSS floats that persist even if you change the order of elements. Floating in general is rather a pain, and CSS3 has means to overcome this rather tedious way of setting positions of elements, which won't work with the PDF module that is in use here, so no CSS3 for that matter.

**Comment by BIIINO** [comment]
@enoversum Yes i know thats the reason why i open the topic :) I think my knowledge is enough to do this. I see in another topics here on github under crater issues with little bit "mess" for this part of code.
